### PR TITLE
Document the variable definition syntax

### LINKED ISSUE #1471: document stack file in references
Currently stacks documentation only mentions `expose:` key and does not mention anything about `variables:`

### PR DESCRIPTION
**Pull request by kke**
Fixes #1471

### CODE REVIEW

File: docs/references/kontena-yml-variables.md [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+
+Define what to do with the value. The default behavior is to set it to local environment using the variable name. You can also use another environment variable name or write the value to the Vault on Kontena Master.
+
+```
+  to:
+    env: MYSQL_USER # set to local env
+    vault: foo-mysql-user # also send to vault
+```
+
+### Only_if
```
Comment (SpComb) [inline docs/references/kontena-yml-variables.md]: These look a bit odd, maybe it would be better to just use fixed-width for these titles

#### `only_if`

**Comment by kke** [comment]
Hmm yes maybe interpolator should escape the output so that it gets through yaml parsing.

**Comment by jakolehm** [comment]
Markdown seems to be broken (at least when viewed with github)?

File: docs/references/kontena-yml-variables.md [hunk 2]
```diff
@@ -1,19 +1,17 @@
 ---
 title: kontena.yml variables
-toc_order: 2
----
```
Comment (SpComb) [inline docs/references/kontena-yml-variables.md]: Lost the YAML header

**Comment by SpComb** [comment]
Replaced by #1531, include the new opto 1.5.3 docs in #1528 ?

**Comment by nevalla** [comment]
@kke needs rebase

**Pull request merged**